mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageResultIterator.java

```diff
@@ -154,7 +159,21 @@ private void readBatch() throws MailboxException {
             range = MessageRange.range(cursor, to);
             break;
         }
-        next = mapper.findInMailbox(mailbox, range, ftype, batchSize);
+        next = mapper.findInMailbox(mailbox, range, ftype, batchSizeFromFetchType(ftype));
+    }
+
+    private int batchSizeFromFetchType(FetchType fetchType) {

```

Comment:
I would have move that code to `FetchBatchSizes`. But maybe that is not a good idea. What do you think ?